lib/charms/prometheus_k8s/v0/prometheus_remote_write.py

```diff
@@ -436,6 +449,14 @@ def _group_name(self, root_path: str, file_path: str, group_name: str) -> str:
         # filter to remove empty strings
         return "_".join(filter(None, group_name_parts))
 
+    def _is_already_modified(self, name: str) -> bool:
+        """Detect whether a group name has already been modified with juju topology."""
+        uuid_regex = "[a-fA-F0-9]{8}-[a-fA-F0-9]{4}-[a-fA-F0-9]{4}-[a-fA-F0-9]{4}-[a-fA-F0-9]{12}"
+        final_regex = ".*_{}_.*_alerts".format(uuid_regex)
```

Comment:
A sick mind could create a false positive here.
I.e. this check doesn't guarantee 100% what it claims to do.
Any chance we could drop this function?

Reply:
No we can't drop it. I think this will create a false positive seldom enough that it should be okay.

Reply:
We all know I love regexps, but there are probably nicer ways to do this. Either just treat hex like hex and use multiple groups for the middle:

```python
r'[\da-f]{8}-([\da-f]{4}-){3}[\da-f]{12}'
```

If we can pull it out, just:
```python
uuid.UUID(str(val)) or ...
``` 

Really, though, I worry about the greedy matchers without a stop. `.*_` is not cool without `?`. Also please use raw strings with regexps.

```python
modified_matcher = re.compile(r'^.*?_[\da-f]{8}-([\da-f]{4}-){3}[\da-f]{12}_.*?alerts$')
if modified_matcher.match(name): ...
```

it's more complex looking from the outset, but less troublesome.

That said, since we know what position in the formatted string this _should_ be in, why don't we split it and check against `uuid.UUID()`? This is ok, and I'm giving it :+1:, but it's a little bit "write-only"

Reply:
Used Ryan's